misc/kafka-util/scripts/summarize_archives.py

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/env python3
+#
+# Copyright Materialize, Inc. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+# Script to read all topic archives and produce a basic statistical summary of the messages
+# contained within
+
+import argparse
+import collections
+import functools
+import io
+import json
+import logging
+import multiprocessing
+import glob
+import os
+import sys
+import typing
+
+import avro.io  # type: ignore
+import avro.schema  # type: ignore
+import pyarrow  # type: ignore
+import pyarrow.fs  # type: ignore
+
+# Setup basic formatting for logging output
+logging.basicConfig(format="%(asctime)s %(levelname)s %(name)s %(message)s")
+log = logging.getLogger("summarize.archive")
+log.setLevel(logging.INFO)
+
+
+def decode(reader: avro.io.DatumReader, msg: bytes) -> typing.Any:
+    """Return the Python object represented by the avro-encoded message."""
+    decoder = avro.io.BinaryDecoder(io.BytesIO(msg))
+    return reader.read(decoder)
+
+
+def summarize_archive(archive: str) -> None:
+    """Print interesting statistics about an archived Kafka topic."""
+
+    with open("schemas.json") as fd:
+        schemas = json.load(fd)
+
+    topic = os.path.splitext(archive)[0]
+    try:
+        key_schema = schemas[f"{topic}-key"]["schema"]
+        value_schema = schemas[f"{topic}-value"]["schema"]
+    except KeyError:
+        log.error(f"Failed to locate schema for topic {topic}")
+        raise
```

Comment:
same comment about error stringification.

Reply:
Yeah, the string interpolation requirement is unfortunate but understandable. I had been passively ignoring this, but I'll fix these up.